### PR TITLE
Tie to the custom docker network for the devcontainers

### PR DESCRIPTION
**Pull request by jimleroyer**
# Summary | Résumé

* Attaching to the custom notify docker network (which should be created in the API project). 
* Refining a few building steps around the devcontainer build.
* Deleting old docker-compose.yml file. 
* Installing a new docker-compose.yml file to get used by the devcontainer.

# Test instructions | Instructions pour tester la modification

Rebuild the devcontainer and make sure you can start the application again. 

Also, having the latest API with related changes, you can now perform the following while the API devcontainer is also up and running:

```
$ curl http://notify-api:6011 
{"current_time_utc": "2024-11-01 20:10:46.583150+00:00", "status": "ok", "commit_sha": "91efa1cb846a152574f6e6a537af43a8527a0114", "build_time": "2024-11-01:19:25:34", "db_version": "0460_new_service_columns"}

$ ping notify-api
PING notify-api (172.21.0.4) 56(84) bytes of data.
64 bytes from gc-notify-notify-api-1.notify-network (172.21.0.4): icmp_seq=1 ttl=64 time=0.127 ms
64 bytes from gc-notify-notify-api-1.notify-network (172.21.0.4): icmp_seq=2 ttl=64 time=0.065 ms
64 bytes from gc-notify-notify-api-1.notify-network (172.21.0.4): icmp_seq=3 ttl=64 time=0.064 ms

$ ping db        
PING db (172.21.0.3) 56(84) bytes of data.
64 bytes from gc-notify-db-1.notify-network (172.21.0.3): icmp_seq=1 ttl=64 time=0.081 ms
64 bytes from gc-notify-db-1.notify-network (172.21.0.3): icmp_seq=2 ttl=64 time=0.069 ms
64 bytes from gc-notify-db-1.notify-network (172.21.0.3): icmp_seq=3 ttl=64 time=0.062 ms

$ ping redis
PING redis (172.21.0.2) 56(84) bytes of data.
64 bytes from gc-notify-redis-1.notify-network (172.21.0.2): icmp_seq=1 ttl=64 time=0.080 ms
64 bytes from gc-notify-redis-1.notify-network (172.21.0.2): icmp_seq=2 ttl=64 time=0.060 ms
64 bytes from gc-notify-redis-1.notify-network (172.21.0.2): icmp_seq=3 ttl=64 time=0.061 ms

$ ping notify-admin
PING notify-admin (172.21.0.5) 56(84) bytes of data.
64 bytes from 1e032ac6fc8e (172.21.0.5): icmp_seq=1 ttl=64 time=0.021 ms
64 bytes from 1e032ac6fc8e (172.21.0.5): icmp_seq=2 ttl=64 time=0.039 ms
64 bytes from 1e032ac6fc8e (172.21.0.5): icmp_seq=3 ttl=64 time=0.036 ms
```

### CODE REVIEW

**Review by whabanks — APPROVED** [review]
LGTM, app runs and communicates with the API container just fine using `host.docker.internal` in the .env with Colima running.

File: .devcontainer/scripts/notify-dev-entrypoint.sh [hunk 1]
```diff
@@ -10,5 +10,5 @@ set -ex
 # Tell git the workspace repository is safe, else upcoming commands will fail.
 git config --global --add safe.directory /workspaces/notification-admin
 
-chown -R vscode:vscode /workspaces/notification-admin
-sudo -u vscode -i -- /usr/local/bin/installations.sh
+# Install and setup dev environment
+installations.sh
```
Comment (jimleroyer) [inline .devcontainer/scripts/notify-dev-entrypoint.sh:14]: I was wondering how this script was called previously but could not find it. Does that look ok if I call it from here? cc @jzbahrai @whabanks @andrewleith

File: .devcontainer/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+version: '3'
```
Comment (jimleroyer) [inline .devcontainer/docker-compose.yml:1]: Having a docker-compose.yml file gives us more options to customize our VSCode devcontainers.

File: .devcontainer/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    #command: /bin/sh -c "while sleep 1000; do :; done"
+    command: sleep infinity
+    expose:
+      - "6012"
+    networks:
+      - notify-network
+
+networks:
+  notify-network:
+    external: true
```
Comment (jimleroyer) [inline .devcontainer/docker-compose.yml:33]: This will attach to the custom network defined in the API project, where it is created.

**Pull request merged**